lib/chef/provisioning/aws_driver/driver.rb

```diff
@@ -570,16 +598,20 @@ def allocate_load_balancer(action_handler, lb_spec, lb_options, machine_specs)
 
       # Update instance list, but only if there are machines specified
       if machine_specs
-        assigned_instance_ids = actual_elb.instances.map { |i| i.instance_id }
-
-        instances_to_add = machine_specs.select { |s| !assigned_instance_ids.include?(s.reference['instance_id']) }
-        instance_ids_to_remove = assigned_instance_ids - machine_specs.map { |s| s.reference['instance_id'] }
+        instances_to_add = 0
```

Comment:
This should be an empty array, rather than an integer:
```
irb(main):003:0> instances_to_add = 0
=> 0
irb(main):004:0> instances_to_add.size
=> 8
irb(main):005:0> instances_to_add = []
=> []
irb(main):006:0> instances_to_add.size
=> 0
```